source/s3_request_messages.c

```diff
@@ -457,30 +458,63 @@ static const struct aws_byte_cursor s_slash_char = AWS_BYTE_CUR_INIT_FROM_STRING
  */
 struct aws_http_message *aws_s3_get_source_object_size_message_new(
     struct aws_allocator *allocator,
-    struct aws_http_message *base_message) {
-    struct aws_http_message *message = NULL;
+    struct aws_http_message *base_message,
+    struct aws_uri *source_uri) {
+
+    struct aws_http_message *message = aws_http_message_new_request(allocator);
     struct aws_byte_buf head_object_host_header;
     AWS_ZERO_STRUCT(head_object_host_header);
 
+    if (message == NULL) {
```

Comment:
do we need to worry about the signing region here today as well? or we are keeping the source must be in the same region as dest behavior?